subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -2332,3 +2332,17 @@ size_t lwm2m_cache_size(const struct lwm2m_time_series_resource *cache_entry)
 	return 0;
 #endif
 }
+
+int lwm2m_set_bulk(const struct lwm2m_res_item res_list[], size_t res_list_size)
+{
+	int ret;
+	for (int i = 0; i < res_list_size; i++) {
```

Comment:
Wouldn't it make sense to take the registry lock before iterating? That way we could set several resources in an "atomic" way.